playground/frontend/lib/modules/editor/components/share_dropdown/share_tabs/share_tabs.dart

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import 'package:flutter/material.dart';
+import 'package:playground/modules/editor/components/share_dropdown/share_tabs/example_tabs.dart';
+import 'package:playground/modules/editor/components/share_dropdown/share_tabs/snippet_tabs.dart';
+import 'package:playground/pages/playground/states/examples_state.dart';
+import 'package:playground/pages/playground/states/playground_state.dart';
+import 'package:provider/provider.dart';
+
+class ShareTabs extends StatelessWidget {
+  final TabController tabController;
+
+  const ShareTabs({super.key, required this.tabController});
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      color: Theme.of(context).backgroundColor,
+      child: Consumer2<PlaygroundState, ExampleState>(
+        builder: (context, playgroundState, exampleState, _) {
+          if (playgroundState.isExampleChanged) {
+            return SnippetTabs(
+              exampleState: exampleState,
+              playgroundState: playgroundState,
+              tabController: tabController,
+            );
+          } else {
+            return ExampleTabs(
```

Comment:
To me this feels more like a special case to be checked with `if`.
The former feels more like the main scenario so it may go after `if ... return`.
But that is a matter of taste.